### PR TITLE
More log traceability

### PR DESCRIPTION
**Pull request by ahgittin**
Many tasks did not correctly get task id and entity ids as part of logging context.  This fixes that.

It also tweaks the format not to include all ancestor entities as part of each log message, just app ID and entity ID; and adds LoggingCategories with messages for common activities (entity creation/deletion, app creation/deletion, task tree).

Format made a little nicer too, but not (yet) for OSGi, just POJO and tests.

### CODE REVIEW

File: core/src/test/java/org/apache/brooklyn/core/entity/ApplicationLoggingTest.java [hunk 1]
```diff
@@ -97,12 +91,13 @@ protected void doStop() {
     public static final class TestEntityWithLoggingImp extends TestEntityImpl implements TestEntityWithLogging {
```
Comment (iuliana) [inline core/src/test/java/org/apache/brooklyn/core/entity/ApplicationLoggingTest.java:91]: I think you intended this class to be named `TestEntityWithLoggingImpl`. The 'L' at the end is misisng.

**Review by iuliana — COMMENTED** [review]
LGTM, just check that class name I mentioned in my comment.

File: logging/logback-includes/src/main/resources/brooklyn/logback-appender-file.xml [hunk 2]
```diff
@@ -23,7 +23,8 @@
     <file>${logging.dir:-./}${logging.basename:-brooklyn}.debug.log</file>
     <append>true</append>
     <encoder>
-      <pattern>%d %-5level %logger{30} [%thread{15}]: %msg%n</pattern>
+      <pattern>%d{ISO8601} %8X{task.id}-%-23X{entity.ids} %-5.5p [%.16t] %-48c{1}  %m%n</pattern>
```
Comment (jcabrerizo) [inline logging/logback-includes/src/main/resources/brooklyn/logback-appender-file.xml:26]: For using UTC timezone the pattern should be
```
<pattern>%d{ISO8601}{UTC}Z %8X{task.id}-%-23X{entity.ids} %-5.5p [%.16t] %-48c{1}  %m%n</pattern>
```

File: logging/logback-includes/src/main/resources/brooklyn/logback-appender-file.xml [hunk 3]
```diff
@@ -44,7 +45,8 @@
     <file>${logging.dir:-./}${logging.basename:-brooklyn}.info.log</file>
     <append>true</append>
     <encoder>
-      <pattern>%d %-5level %logger{30} [%thread{15}]: %msg%n</pattern>
+      <pattern>%d{ISO8601} %8X{task.id}-%-23X{entity.ids} %-5.5p %-48c{1}  %m%n</pattern>
```
Comment (jcabrerizo) [inline logging/logback-includes/src/main/resources/brooklyn/logback-appender-file.xml:48]: Same as above

File: logging/logback-includes/src/main/resources/brooklyn/logback-appender-jclouds.xml [hunk 4]
```diff
@@ -29,7 +29,8 @@
                 <file>${logging.dir:-./}${logging.basename:-brooklyn}.${jcloudsPersistSwitch}.log</file>
                 <append>true</append>
                 <encoder>
-                    <pattern>%d %-5level %logger{30} [%thread{15}]: %msg%n</pattern>
+                    <pattern>%d{ISO8601} %-5.5p %c{1}  %m%n</pattern>
```
Comment (jcabrerizo) [inline logging/logback-includes/src/main/resources/brooklyn/logback-appender-jclouds.xml:32]: Same as above

File: logging/logback-includes/src/main/resources/brooklyn/logback-appender-stdout.xml [hunk 5]
```diff
@@ -21,7 +21,9 @@
 
   <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
     <encoder>
-      <pattern>%d %-5level %X{task.id}-%X{entity.ids} %msg%n</pattern>
+<!--      <pattern>%d %-5level %X{task.id}-%X{entity.ids} %msg%n</pattern> -->
+<!--      <pattern>%d{ISO8601}{UTC}Z %X{task.id}-%X{entity.ids} %-5.5p %c{1.} [%.16t] %m%n</pattern>-->
+      <pattern>%d{ISO8601} %-5.5p %-48c{1}  %m%n</pattern>
```
Comment (jcabrerizo) [inline logging/logback-includes/src/main/resources/brooklyn/logback-appender-stdout.xml:26]: Same as above

File: test-support/src/main/java/org/apache/brooklyn/test/LogWatcher.java [hunk 6]
```diff
@@ -186,13 +187,7 @@ public LogWatcher(Iterable<String> loggerNames, ch.qos.logback.classic.Level log
...
         };
 
-        // The code below makes the assumption that the (test) logger configuration has a console appender
-        // for root, with a pattern layout encoder, and re-uses its encoder pattern.
-        // This is (at time of writing) as defined in logback-appender-stdout.xml.
-        final Appender<ILoggingEvent> appender = lc.getLogger("ROOT").getAppender("STDOUT");
-        final ConsoleAppender<?> stdout = ConsoleAppender.class.cast(appender);
-        final PatternLayoutEncoder stdoutEncoder = PatternLayoutEncoder.class.cast(stdout.getEncoder());
-        ple.setPattern(stdoutEncoder.getPattern());
+        ple.setPattern("%d{ISO8601} %8X{task.id}-%-23X{entity.ids} %-5.5p %-48c{1}  %m%n");
```
Comment (jcabrerizo) [inline test-support/src/main/java/org/apache/brooklyn/test/LogWatcher.java:190]: Same as above

File: utils/test-support/src/main/resources/brooklyn/logback-appender-file.xml [hunk 7]
```diff
@@ -23,7 +23,8 @@
     <file>${logging.dir:-./}${logging.basename:-brooklyn}-tests.log</file>
     <append>true</append>
     <encoder>
-      <pattern>%d %-5level %logger{30} [%thread{15}]: %msg%n</pattern>
+      <pattern>%d{ISO8601} %8X{task.id}-%-23X{entity.ids} %-5.5p [%.16t] %-48c{1}  %m%n</pattern>
```
Comment (jcabrerizo) [inline utils/test-support/src/main/resources/brooklyn/logback-appender-file.xml:26]: Same as above

**Review by jcabrerizo — CHANGES_REQUESTED** [review]
LGTM but please, use UTC for the logs to keel all config consistence

**Review by jcabrerizo — APPROVED** [review]

**Pull request merged**